partner_preview_scripts/ValidationLab.py

```diff
@@ -3,432 +3,472 @@
+            .select(
+                # constants:
+                "datasourceId",
+                "datatypeId",
+                "studyOverview",
+                "projectId",
+                "releaseDate",
+                "releaseVersion",
+                # Evidence:
+                "targetFromSourceId",
+                "diseaseFromSourceMappedId",
+                "diseaseFromSource",
+                # Assessments:
+                "resourceScore",
+                "assessment",
+                "assays",
+                # Cell lines:
+                f.array(
+                    f.struct(
+                        f.col("cellLineName").alias("name"),
+                        f.col("id").alias("id"),
+                        f.col("tissueFromSource").alias("tissue"),
+                        f.col("tissueId"),
+                    )
+                ).alias("diseaseCellLines"),
+                "biomarkerList",
+                # Primary project:
+                "primaryProjectHit",
+                "primaryProjectId",
+            )
+        )
 
-    # The biomarker columns are unstacked into one single 'biomarkers' column:
-    biomarker_unstack = f"""stack({len(biomarkers_in_data)}, {", ".join([f"'{x}', {x}" for x in biomarkers_in_data])}) as (biomarker_name, biomarkers)"""
 
-    validation_lab_cell_lines = (
-        biomarkers
-        # Selecting cell line name, cell line annotation and applyting the stacking expression:
-        .select(
-            f.col("cellName"),
-            "diseaseCellLines",
-            f.expr(biomarker_unstack),
+@dataclass
+class ValidationLabProjectParser:
+    diseaseFromSourceMappedId: str
+    diseaseFromSource: str
+    tissueId: str
+    experimentDataFile: str
+    biomarkerDataFile: str
+    studyOverview: str
+    primaryProjectId: str
+    releaseVersion: str
+    releaseDate: str
+    excludeStudy: bool
+
+    def parse_evidence(
+        self: ValidationLabProjectParser,
+        spark: SparkSession,
+        input_path: str,
+        assays: list,
+    ) -> DataFrame:
+        logger = logging.getLogger(__name__)
+
+        validation_input_file = f"{input_path}/{self.experimentDataFile}"
+        biomarker_file = f"{input_path}/{self.biomarkerDataFile}"
+
+        logger.info(f"Reading project data: {validation_input_file}")
+        logger.info(f"Reading biomarker data: {biomarker_file}")
+
+        # Reading data:
+        raw_experiment_data = spark.read.option("multiline", True).csv(
+            validation_input_file, sep="\t", header=True
         )
-        # Filter out all null biomarkers:
-        .filter(
-            (f.col("biomarkers").isNotNull())
-            &
-            # Following the request of the validation lab, we are removing CRIS biomarker annotation:
-            (f.col("biomarker_name") != "CRIS_subtype")
+        # First round of processing:
+        processed_experiment = self._process_raw_evidence(raw_experiment_data, assays)
```

Comment:
Maybe this distinction between `_process_raw_evidence` and `parse_evidence` is too artificial? I can see that `_process_raw_evidence` are just light transformations, but I don't know if the concern is separated enough to be in an indepedent function.

Reply:
Probably not. I was kind of struggling finding the right level of modularity. I'll take a look at how it could be made sensible. I just really don't like to write long functions (not that I was writing tests).